### PR TITLE
Add basic leveled logging

### PR DESCRIPTION
**Pull request by aboodman**
Addresses rocicorp/replicache#35 for the flutter repo.

### CODE REVIEW

**Comment by aboodman** [comment]
I don't know if this is how you'd like this to be done, but here's a first cut.

File: lib/replicache.dart [hunk 1]
```diff
@@ -145,6 +146,15 @@ class Replicache implements ReadTransaction {
     return rep;
   }
 
+  /// Sets the level of verbosity Replicache should log at.
+  static setLogLevel(LogLevel level) {
```
Comment (arv) [inline lib/replicache.dart:150]: [taste] Use a getter and a setter.

File: lib/replicache.dart [hunk 2]
```diff
@@ -145,6 +146,15 @@ class Replicache implements ReadTransaction {
...
+    globalLogLevel = level;
```
Comment (arv) [inline lib/replicache.dart:151]: [cbb] It is strange to have both a static property and a global.

File: lib/replicache.dart [hunk 3]
```diff
@@ -145,6 +146,15 @@ class Replicache implements ReadTransaction {
...
+    _staticInvoke('', 'setLogLevel',
+      {LogLevel.debug:'debug',
+      LogLevel.info:'info',
+      LogLevel.error:'error'}[level]);
```
Comment (arv) [inline lib/replicache.dart:155]: [taste] Use trailing comma and auto format will make this look nicer.

File: sample/redo/lib/main.dart [hunk 4]
```diff
@@ -17,6 +17,7 @@ String addPrefix(int id) => '$prefix$id';
 class MyApp extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
+    Replicache.setLogLevel(LogLevel.debug);
```
Comment (arv) [inline sample/redo/lib/main.dart:20]: [cbb] Seems better to do this outside the build method.

Comment (aboodman) [reply sample/redo/lib/main.dart:20]: Crap I didn't mean to check this in.

**Pull request merged**